libraries/lib-music-information-retrieval/MirDsp.cpp

```diff
@@ -0,0 +1,138 @@
+#include "MirDsp.h"
+#include "FFT.h"
+#include "MirAudioReader.h"
+#include "MirTypes.h"
+#include "MirUtils.h"
+#include "StftFrameProvider.h"
+
+#include <cassert>
+#include <cmath>
+#include <numeric>
+
+namespace MIR
+{
+namespace
+{
+float GetNoveltyMeasure(
+   const std::vector<float>& prevPowSpec, const std::vector<float>& powSpec)
+{
+   auto k = 0;
+   return std::accumulate(
+      powSpec.begin(), powSpec.end(), 0., [&](float a, float mag) {
+         // Half-wave-rectified stuff
+         return a + std::max(0.f, mag - prevPowSpec[k++]);
+      });
+}
+
+std::vector<float>
+GetMovingAverage(const std::vector<float>& x, double sampleRate)
+{
+   constexpr auto smoothingWindowDuration = 0.2;
+   // An odd number.
+   const int M = std::round(smoothingWindowDuration * sampleRate / 4) * 2 + 1;
+   const auto window = GetNormalizedHann(2 * M + 1);
```

Comment:
This use of the normalized Hann window is only as a moving-average kernel -- spectral properties are not important.  Am I right?

Reply:
I suppose it is in some way. The average of the signal is like its DFT at 0Hz. The window function reduces the amount of leakage of non-zero components in the 0Hz measurement.

Reply:
... but it's not meant for improved frequency-domain analysis, if that's what you meant.